### PR TITLE
[xpu] Update SDPA OneDNN graph for training performance

### PR DESCRIPTION
**Pull request by LuFinch**
Split from PR https://github.com/pytorch/pytorch/pull/162454. This PR only changes files under XPU mkldnn for review quickly.

OneDNN v3.12 supports GQA/dropout and refactor the SDPA pattern so that they can get more precise information and do optmization in v3.13. This PR integrates GQA/dropout and aligns with the new graph pattern.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01

### CODE REVIEW

**Comment by LuFinch** [comment]
@guangyey @EikanWang Could you help review?

**Review by Skylion007 — APPROVED** [review]

**Review by EikanWang — COMMENTED** [review]
Any test plan?

**Comment by LuFinch** [comment]
> Any test plan?

@EikanWang 

I run a round dynamo benchmark fp16 training performance test on B580.

Below table is geomean ratio (with this PR vs without this PR)
|          | HuggingFace | Timms | TorchBench |
|----------|-------------|-------|------------|
| Eager    | 0.941       | 1.008 | 1.002      |
| Inductor | 0.853       | 1.043 | 0.987      |

OneDNN helps minimize the performance gap between OneDNN and SYCLTLA in v3.13. The ratio is only 50%~60% last time. 

Even though it is still slower than SYCLTLA SDPA backward and brings regression to huggingface,  AE team asks whether we could merge OneDNN SDPA backward, because they are testing on Windows and SYCLTLA doesn't support Windows.

I think we should merge this PR and users can switch between OVERRIDEABLE and FLASH_ATTENTION backend accroding to measured performance by themself. Do you think this is a reasonable to merge this PR?

**Review by EikanWang — APPROVED** [review]

File: aten/src/ATen/native/mkldnn/xpu/Attention.cpp
Comment (guangyey) [inline aten/src/ATen/native/mkldnn/xpu/Attention.cpp:319]: Is this still required?

Comment (LuFinch) [reply aten/src/ATen/native/mkldnn/xpu/Attention.cpp:319]: Will remove this check in https://github.com/pytorch/pytorch/pull/162454.  This PR only aims at mkldnn/xpu changes about onednn graph api.

File: aten/src/ATen/native/mkldnn/xpu/Attention.cpp [hunk 1]
```diff
@@ -335,6 +335,9 @@ _scaled_dot_product_fused_attention_overrideable_xpu(
       batch_size, num_head_q, seq_len_q, head_dim_v};
   alloc_with_matching_layout(query, output, output_shape);
   at::Tensor logsumexp, debug_attn_mask; // not supported
+  // rng not used
```
Comment (guangyey) [inline aten/src/ATen/native/mkldnn/xpu/Attention.cpp:338]: Now it is used in line 360/361, so should we remove this comment

**Review by guangyey — APPROVED** [review]

File: aten/src/ATen/native/mkldnn/xpu/Attention.cpp (hunk 1 above)
Comment (LuFinch) [reply aten/src/ATen/native/mkldnn/xpu/Attention.cpp:338]: same as above
